pingcap/tidb
# sessionctx: Set foreign_key_checks = OFF

### What problem does this PR solve? 

Partially address https://github.com/pingcap/tidb/issues/8349 by making metadata clearer that foreign keys are not supported.

### What is changed and how it works?

This does two things:
1. Changes the default for `foreign_key_checks` to `OFF` (differs from MySQL default)
2. Shows a warning (and refuses to change) `foreign_key_checks` to `ON`.  Since this is not supported by TiDB.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

I've manually verified it works.  It does not actively change code, since it is currently an unused option - but I can add unit tests if desired.

Code changes

 - Minimal

Side effects

 - Breaking backward compatibility (for example, some exotic loaders may check for a warning on reload.)

Related changes

 - Need to update the documentation (default differences from mysql)
 - Need to update the `tidb-ansible` repository (I'm not sure - I don't think it is required).


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pingcap/tidb/8358)
